### DISCUSSION TITLE
App Store Connect API 1.7 - Features Requests and Usage Suggestions

### DISCUSSION THREAD

**Discussion by joshdholtz** [fastlane/fastlane#19862] [Ideas]
Version 1.7 of the App Store Connect API dropped on January 25, 2021.

👉 I would love to know everybody would like to use this! Would you expect this to be part of `deliver`? Would you use `spaceship` directly for any of this? Do you think a whole new tool is necessary?

Below is a list of all the new features that can be accessed 👇

✅ AppCustomProduct
✅ AppEvent
✅ AppStoreVersionExperiment
✅ AppStoreVersionPromotion

❌ AppStoreVersionSubmission (deprecated)

✅ ReviewSubmission
✅ ReviewSubmissionItems

Fire those ideas at me!

**Comment by Patrick-Kladek** [comment]
I want to support custom product pages in AppStore eg different screenshots.
Right now the workflow is super easy. Create screenshots and use frameit. Then simply upload it to appstoreconnect with deliver.

Would be great to simply create a new folder with the same structure as before but different metadata/screenshots and deliver is able to upload it as second variant to appstoreconnect.

Doing this manually when you support 14 languages is not a fun task especially for an A/B Test.

**Reply by arshiacont** [reply]
Dropping on this rather late… I believe as many of these new features are being adopted, their support in Deliver or Spaceship would be highly appreciated.

is it currently possible to update AppEvents directly from Spaceship today?! Didn’t find anything on docs.

**Reply by keesvandieren** [reply]
Herewith some feature suggestions from our side:

High prio: Support for Custom Product Page  (screenshots for localisations)
Medium prio: Screenshots for Product Page Optimization (A/B testing)

**Reply by Huaba93** [reply]
Is the AppStoreVersionExperiment feature already available?

**Reply by dsilkensen** [reply]
Is there a general walkthrough for how to setup Spaceships ConnectAPI to use the API calls?

I'd love to look into getting it to work for In-App Events (appEvents), but I'm not sure if there's a walkthrough on how to setup the calls within Spaceship.

**Reply by jianwen-wang** [reply]
Until now, fastlane does not support uploading screenshots to custom product pages. Is there another good way to accomplish this feature?

**Reply by kashitamang** [reply]
Would really really love to see support for AppEvents through Fastlane. I have been able to validate the simple HTTP calls using a GUI. Can this be an added feature?!
